src/extension.ts

```diff
@@ -120,19 +120,89 @@ export async function activate(context: vscode.ExtensionContext): Promise<vscode
 					await vscode.window.showInformationMessage(messages.graphEngine.noViolationsFound);
 					return;
 				});
+				const methodLevelTarget: string = await targeting.getSelectedMethod();
+				// Pull out the file from the target and use it to identify the project directory.
+				const currentFile: string = methodLevelTarget.substring(0, methodLevelTarget.lastIndexOf('#'));
+				const projectDir: string = targeting.getProjectDir(currentFile);
+
 				return _runAndDisplayDfa(context, {
 					commandName: Constants.COMMAND_RUN_DFA_ON_SELECTED_METHOD,
 					outputChannel
-				}, customCancellationToken);
+				}, customCancellationToken, methodLevelTarget, projectDir);
 			});
 		}
 	});
-	context.subscriptions.push(runOnActiveFile, runOnSelected, runDfaOnSelectedMethod, removeDiagnosticsOnActiveFile, removeDiagnosticsOnSelectedFile, removeDiagnosticsInRange);
+
+	const runDfaOnWorkspaceCmd = vscode.commands.registerCommand(Constants.COMMAND_RUN_DFA, async () => {
+		await _runDfa(context, outputChannel);
+	});
+	context.subscriptions.push(runOnActiveFile, runOnSelected, runDfaOnSelectedMethodCmd, runDfaOnWorkspaceCmd, removeDiagnosticsOnActiveFile, removeDiagnosticsOnSelectedFile, removeDiagnosticsInRange);
 	TelemetryService.sendExtensionActivationEvent(extensionHrStart);
 	outputChannel.appendLine(`Extension sfdx-code-analyzer-vscode activated.`);
 	return Promise.resolve(context);
 }
 
+async function _runDfa(context: vscode.ExtensionContext, outputChannel: vscode.LogOutputChannel) {
+	const choice = await vscode.window.showQuickPick(
```

Comment:
Does this prompt show up every time - even when you haven't ran before? Ideally it should only show up when we have ran already with violations.

Reply:
As discussed in the standup, took care of this to show only when cache exists.